### PR TITLE
Mean absolute error layer and L1 norm layer

### PR DESCRIPTION
**Pull request by timmoon10**
The mean absolute error layer is necessary to replace some deprecated functionality used in the CANDLE Pilot2 models. It also makes progress toward #346. I've taken the liberty of making some cosmetic changes to the tensor dimension error checking in the loss layers.

### CODE REVIEW

**Comment by timmoon10** [comment]
The unit tests achieve the expected values and gradient checks pass.

File: src/layers/loss/l1_norm.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+#include "lbann/layers/loss/l1_norm.hpp"
+
+namespace lbann {
+
+namespace {
+
+void local_fp_cpu(const AbsMat& local_input,
+                  AbsMat& local_contribution) {
+#pragma omp parallel for
```
Comment (bvanessen) [inline src/layers/loss/l1_norm.cpp]: Can we please change this to match the new style of LBANN_OMP_PARALLEL_FOR

File: src/layers/loss/l1_norm.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    local_contribution(0, col) = sum;
+  }
+}
+
+void local_bp_cpu(const AbsMat& local_input,
+                  const AbsMat& local_gradient_wrt_output,
+                  AbsMat& local_gradient_wrt_input) {
+  constexpr DataType zero = 0;
+#pragma omp parallel for collapse(2)
```
Comment (bvanessen) [inline src/layers/loss/l1_norm.cpp]: LBANN_OMP_PARALLEL_FOR_COLLAPSE2

File: src/layers/loss/mean_absolute_error.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                  const AbsMat& local_prediction,
+                  const AbsMat& local_ground_truth,
+                  AbsMat& local_contribution) {
+
+  // Useful constants
+  const auto& local_height = local_prediction.Height();
+  const auto& local_width = local_prediction.Width();
+
+  // Compute local contribution to mean absolute error
+#pragma omp parallel for
```
Comment (bvanessen) [inline src/layers/loss/mean_absolute_error.cpp]: LBANN_OMP_PARALLEL_FOR

File: src/layers/loss/mean_absolute_error.cpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                  AbsMat& local_gradient_wrt_prediction,
+                  AbsMat& local_gradient_wrt_ground_truth) {
+                                                                       
+  // Useful constants
+  const DataType scale = DataType(1) / height;
+  const El::Int local_height = local_prediction.Height();
+  const El::Int local_width = local_prediction.Width();
+
+  // Compute gradients
+#pragma omp parallel for collapse(2)
```
Comment (bvanessen) [inline src/layers/loss/mean_absolute_error.cpp]: LBANN_OMP_PARALLEL_FOR_COLLAPSE2

**Pull request merged**